ndlarflow/h5_to_root_ndlarflow.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+import awkward as awk
+
+import uproot as ur
+import h5py as h5
+
+from h5flow.data import dereference
+import h5flow
+import os
+import sys
+
+# Refactored version of h5 to ROOT conversion, Bruce Howard - 2025
+# initial scripts are now saved in e.g. _minirun6_2.py and _minirun6_3.py versions e.g. and thanks to Richie Diurba and any others who made these scripts
+
+# NOTE for more on uproot TTree writing see the below or the uproot documentation
+# see https://stackoverflow.com/questions/72187937/writing-trees-number-of-baskets-and-compression-uproot
+
+# Main function with command line settable params
+def printUsage():
+    print('python h5_to_root_ndlarflow_ndmc_more_refactor.py FileList IsData IsFinalHits OutName')
+    print('-- Parameters')
+    print('FileList    [REQUIRED]:                           comma separated set of files to convert - note it will be one output')
+    print('IsData      [OPTIONAL, DEFAULT = 0, is MC]:       1 = Data, otherwise = MC')
+    print('IsFinalHits [OPTIONAL, DEFAULT = 0, prompt hits]: 1 = use "final" hits, otherwise = "prompt"')
+    print('OutName     [OPTIONAL, DEFAULT = input[0]+stuff]: string for an output file name if you want to override. Note that default writes to current directory.')
```

Comment:
For the `OutName` info, can we replace "stuff" with the actual filename extension change? 

By default, for `input.hdf5`,  the output has the name `input.hdf5_hits.root`. Should the default instead be `input.hdf5_hits_uproot.root`, so it is then clear that the ROOT macro needs to be run on this as well?

We should add info to the usage help that the extra ROOT macro step is needed, along with an example run command.

Reply:
Yes and agreed!